### DISCUSSION TITLE
Is it possible to host multiple universal nuxt pages on one server ?

### DISCUSSION THREAD

**Discussion by duderion** [nuxt/nuxt#17171] [Questions]
Hi Nuxt People,

In our env we set a shopId to define where our fetch takes its data from

The line of code for the fetch is

`const { data: brands } = await $fetch(`${config.public.apiBaseUrl}/shop/${config.public.shopId}/brands`);`

One nuxt instance uses shopID 1, one uses shopID 5

What happens now is that the shopID on shop1 changes to 5 on reload.
so shop1 shows the brands of shop 5.

any ideas ?

Adrian

**Comment by bros4president** [comment]
Hi! Could you describe in more detail:
- How are you generating the Nuxt instances?
- Where are you running them?
- What do you mean by reload? (simply pressing F5 in browser?)

**Reply by duderion** [reply]
Hi Luis

- I generate them with yarn build (nuxi in the background)
- I run them on a dedicated server, deployed with forge, pm2 as runner (same effect with pure node), side by side in different directories
- f5 yes. I go to the startpage, all fine, I go to brand page all fine, I press f5 on the brands page and see the brands of the other page (it’s using its shopID) so the configurations seem to clash


I hope I could answer your questions

cheers and thank you

Adrian

**Reply by bros4president** [reply]
Alright Adrian, the following is just a guess based on a similar problem I'm having (issue #6090).

Basically each Nuxt output is designed to be run in an isolated environment. It stores certain variables in `globalThis`.

In my case the static paths were clashing. In your case I'm not completely sure, but at least looking inside `.output/server/chunks/server.mjs`  (filename and content might change depending on your nitro preset), I can see that `$fetch` is also being tied to global state. The reason I'm not sure this is the root of your problem is that even with a shared `$fetch`, changing the url accordingly should work.

Looking inside `.output/server/chunks/aws-lambda.mjs` (my preset is aws-lambda, you probably have a differente filename, but it's the one containing the app handler) I can see that `runtimeConfig` obtains values from `process.env`, which could be the actual root of your problem.

In my issue, #6090 , @danielroe suggests to run different projects in workers to isolate their environment. I haven't had the time to try that solution due to prioritzing other features, so I can't help you further.

May I ask, why are you creating separate Nuxt .outputs from the same project? In my case I need different `/pages` directories, but with the information I have about yours, why not just save the shopId in a store?
